### PR TITLE
Get earliest version by Catalog Client.

### PR DESCRIPTION
**Pull request by iriza7**
* Update CatalogVersionRequest class.

* Get earliest version by Catalog Client.

* Update Unit tests.

Relates-To: OLPEDGE-918.

Signed-off-by: Bautista, Iryna <ext-iryna.bautista@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts [hunk 1]
```diff
@@ -74,7 +74,32 @@ export class CatalogClient {
...
+            HRN.fromString(this.hrn),
+            abortSignal
+        ).catch(error => Promise.reject(error));
+        const earliestVersion = await MetadataApi.minimumVersion(
+            builder,
+            {
+                billingTag: request.getBillingTag()
+            }
+        );
+        return earliestVersion.version;
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts]: Please return Promise with version or reject promise if "If the catalog doesn't contain any versions"

File: @here/olp-sdk-dataservice-read/lib/client/CatalogVersionRequest.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ import { validateBillingTag } from "..";
 export class CatalogVersionRequest {
     private startVersion?: number;
     private endVersion?: number;
+    private earliestVersion?: number;
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/CatalogVersionRequest.ts]: Please create separate request class for that method: CatalogEarliestVersionRequest

File: @here/olp-sdk-dataservice-read/lib/client/CatalogEarliestVersionRequest.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * License-Filename: LICENSE
+ */
+
+import { validateBillingTag } from "..";
+
+/**
+ * A class that prepare information about catalog earliest version.
+ */
+export class CatalogEarliestVersionRequest {
+    private earliestVersion?: number;
```
Comment (andescu) [inline @here/olp-sdk-dataservice-read/lib/client/CatalogEarliestVersionRequest.ts]: Looking at the REST API I don't see any version you need to pass.
What is it used for?

File: @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts [hunk 4]
```diff
@@ -74,7 +75,34 @@ export class CatalogClient {
...
-     * Get the latest version of the catalog.
+     * Returns minimum version for the given catalog. If the catalog doesn't contain any versions error will be returned.
+     * @param request Is [[CatalogEarliestVersionRequest]] instance and has ((getEarliestVersion)) method.
+     * @param abortSignal A signal object that allows you to communicate with a request (such as a Fetch)
+     * and abort it if required via an AbortController object.
+     *  @see https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal
+     *  @returns A promise of the http response that contains the pminimum version for the given catalog.
+     */
+    public async getEarliestVersion(
+        request: CatalogEarliestVersionRequest,
```
Comment (andescu) [inline @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts]: This should share the same request structure as the getLatestVersion() and the getVersions() method.

Comment (iriza7) [reply @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts]: fixed

File: @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts [hunk 5]
```diff
@@ -74,7 +74,34 @@ export class CatalogClient {
...
+            HRN.fromString(this.hrn),
+            abortSignal
+        ).catch(error => Promise.reject(error));        
+
+        const earliestVersion = await MetadataApi.minimumVersion(
+            builder,
+            {
+                billingTag: request.getBillingTag()
+            }
+        ).catch(err => Promise.reject(`The catalog doesn't contain any versions, Error: ${err}`));
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/client/CatalogClient.ts]: Please change to:  Promise.reject(`Error getting earliest catalog version: ${err}`));

**Pull request closed without merging**